File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/FlingFragment.java
package com.pennapps.labs.pennmobile;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.crashlytics.android.Crashlytics;
import com.crashlytics.android.answers.Answers;
import com.crashlytics.android.answers.ContentViewEvent;
import com.pennapps.labs.pennmobile.adapters.FlingRecyclerViewAdapter;
import com.pennapps.labs.pennmobile.api.Labs;
import com.pennapps.labs.pennmobile.classes.FlingEvent;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import io.fabric.sdk.android.Fabric;
import rx.functions.Action1;

public class FlingFragment extends Fragment {

    @BindView(R.id.fling_fragment_recyclerview) RecyclerView flingFragmentRecyclerView;
    private Unbinder unbinder;

    List<FlingEvent> flingEvents;

    public FlingFragment() {
        // Required empty public constructor
    }

    public static FlingFragment newInstance() {
        FlingFragment fragment = new FlingFragment();
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Fabric.with(getContext(), new Crashlytics());
        Answers.getInstance().logContentView(new ContentViewEvent()
                .putContentName("Spring Fling")
                .putContentType("App Feature")
                .putContentId("7"));
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_fling, container, false);
        unbinder = ButterKnife.bind(this, view);
        Labs labs = MainActivity.getLabsInstance();
        labs.getFlingEvents().subscribe(new Action1<List<FlingEvent>>() {
            @Override
            public void call(final List<FlingEvent> flingEvents) {
                getActivity().runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        flingFragmentRecyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
                        flingFragmentRecyclerView.setAdapter(new FlingRecyclerViewAdapter(getContext(), flingEvents));
                    }
                });
            }
        }, new Action1<Throwable>() {
            @Override
            public void call(Throwable throwable) {
                getActivity().runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(getActivity(), "Error: Could not retrieve Spring Fling schedule", Toast.LENGTH_LONG).show();
                    }
                });
            }
        });
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
        getActivity().setTitle(R.string.spring_fling);
//        ((MainActivity) getActivity()).setNav(R.id.nav_fling);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
    }

    @Override
    public void onDetach() {
        super.onDetach();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/FlingFragment.kt
package com.pennapps.labs.pennmobile

import android.net.Uri
import android.os.Bundle
import androidx.browser.customtabs.CustomTabsIntent.Builder
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import android.view.*
import android.widget.Toast
import com.google.firebase.analytics.FirebaseAnalytics
import com.pennapps.labs.pennmobile.adapters.FlingRecyclerViewAdapter
import com.pennapps.labs.pennmobile.databinding.FragmentFlingBinding


class FlingFragment : Fragment() {

    private lateinit var mActivity: MainActivity

    private var _binding : FragmentFlingBinding? = null
    private val binding get() = _binding!!

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
        mActivity = activity as MainActivity
        mActivity.supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val bundle = Bundle()
        bundle.putString(FirebaseAnalytics.Param.ITEM_ID, "7")
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, "Spring Fling")
        bundle.putString(FirebaseAnalytics.Param.ITEM_CATEGORY, "App Feature")
        FirebaseAnalytics.getInstance(mActivity).logEvent(FirebaseAnalytics.Event.VIEW_ITEM, bundle)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.fling_menu, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle presses on the action bar items
        when (item.itemId) {
            android.R.id.home -> {
                mActivity.onBackPressed()
                return true
            }
            R.id.fling_raffle -> {
                val url = "https://docs.google.com/forms/d/e/1FAIpQLSexkehYfGgyAa7RagaCl8rze4KUKQSX9TbcvvA6iXp34TyHew/viewform"
                val builder = Builder()
                val customTabsIntent = builder.build()
                customTabsIntent.launchUrl(mActivity, Uri.parse(url))
            }
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        _binding = FragmentFlingBinding.inflate(inflater, container, false)
        val view = binding.root
        val labs = MainActivity.studentLifeInstance
        labs.flingEvents.subscribe({ flingEvents ->
            activity?.runOnUiThread {
                binding.flingFragmentRecyclerview.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
                binding.flingFragmentRecyclerview.adapter = FlingRecyclerViewAdapter(context, flingEvents)
            }
        }, { activity?.runOnUiThread { Toast.makeText(activity, "Could not retrieve Spring Fling schedule", Toast.LENGTH_LONG).show() } })
        return view
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
        mActivity.supportActionBar?.setDisplayHomeAsUpEnabled(false)
    }

    override fun onResume() {
        super.onResume()
        val mActivity : MainActivity? = activity as MainActivity
        mActivity?.removeTabs()
        mActivity?.setTitle(R.string.spring_fling)
    }
}
